pkg/collector/corechecks/cluster/kubernetes_eventbundle.go

```diff
@@ -56,17 +59,27 @@ func (k *kubernetesEventBundle) addEvent(event *v1.Event) error {
 
 	k.countByAction[fmt.Sprintf("**%s**: %s\n", event.Reason, event.Message)] += int(event.Count)
 	k.readableKey = fmt.Sprintf("%s %s", event.InvolvedObject.Name, event.InvolvedObject.Kind)
+
+	if event.InvolvedObject.Kind == "Node" || event.InvolvedObject.Kind == "Pod" {
+		k.hostname = event.Source.Host
+		clusterName := clustername.GetClusterName()
```

Comment:
To avoid calling that every time, I'd store the nodenode in `kubernetesEventBundle` and append the clustername in  `formatEvents()` instead. This way, you can query it once before iterating on the bundle slice and pass it as a method argument